src/test/java/TrainController/TrainControllerTest.java

```diff
@@ -0,0 +1,285 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package traincontroller;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+/**
+ *
+ * @author Didge
+ */
+public class TrainControllerTest {
+    
+//    public TrainControllerTest() {
+//    }
+//    
+//    @BeforeClass
+//    public static void setUpClass() {
+//    }
+//    
+//    @AfterClass
+//    public static void tearDownClass() {
+//    }
+//    
+//    @Before
+//    public void setUp() {
+//    }
+//    
+//    @After
+//    public void tearDown() {
+//    }
+
+//    /**
+//     * Test of main method, of class TrainController.
+//     */
+//    @Test
+//    public void testMain() throws Exception {
+//        System.out.println("main");
+//        String[] args = null;
+//        TrainController.main(args);
+//        // TODO review the generated test code and remove the default call to fail.
+//        fail("The test case is a prototype.");
+//    }
+
+//    /**
+//     * Test of runTest method, of class TrainController.
+//     */
+//    @Test
+//    public void testRunTest() throws Exception {
+//        System.out.println("runTest");
+//        int period = 0;
+//        TrainController.runTest(period);
+//        // TODO review the generated test code and remove the default call to fail.
+//        fail("The test case is a prototype.");
+//    }
+
+    /**
+     * Test of computeSafeBrake method, of class TrainController.
+     */
+    @Test
+    public void testComputeSafeBrake() {
+        System.out.println("computeSafeBrake");
+        TrainController instance = new TrainController(true);
```

Comment:
As a rule of thumb, you probably shouldn't be using different constructors for testing vs normal usage.

The goal is to run everything the way it will normally be used, then test the outputs to make sure they're what you expect. By having different constructors, it's a different set of code that actually executes, so the real constructor isn't really tested at all.